task-runner/task_runner/task_request_handler.py

```diff
@@ -374,13 +376,29 @@ def _setup_working_dir(self, task_dir_remote) -> str:
             "'_setup_working_dir' called without a task ID.")
 
         task_workdir = os.path.join(self.workdir, self.task_id)
+        sim_workdir = os.path.join(task_workdir, "sim_dir")
 
         if os.path.exists(task_workdir):
             logging.info("Working directory already existed: %s", task_workdir)
             logging.info("Removing directory: %s", task_workdir)
             shutil.rmtree(task_workdir)
 
         os.makedirs(task_workdir)
+        os.makedirs(sim_workdir)
+
+        # Download the workspace folder first so the files can be overwriten
+        # by the task files
+        for input_resource in self.input_resources:
+            # assume its a file if it has a '.'
+            if '.' in input_resource.split('/')[-1]:
+                file_name = input_resource.split('/')[-1]
```

Comment:
The `file_name` should be retrieved before the `if` clause to avoid repeating code.

Reply:
That's true

Also this part will be redone with the new input_resources path format so it won't have to make assumptions to distinguish between tasks and workspaces.